plone/schema/email.py

```diff
@@ -23,6 +17,78 @@ class InvalidEmail(ValidationError):
     __doc__ = _("""The specified email is not valid.""")
 
 
+def _isemail(value):
+    r"""Is this a valid email?
+
+    https://www.regular-expressions.info/email.html has some hints on how to
+    check for a valid email with regular expressions.  It also gives reasons
+    for why you may *not* want to use them.  A too simple regex will work for
+    most cases, but may give false negatives: it will treat a rare but valid
+    email address as invalid.  A complex regex may still allow some invalid
+    email addresses, and is hard to debug in case of errors.
+
+    Originally we had this regex, unchanged between 2013 and 2024:
+
+        import re
+        _isemail = r"[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}"
+        _isemail = re.compile(_isemail).match
+
+    Problems:
+
+    1. It does not allow apostrophes.
+       Example: o'hara@example.org
+       See: https://github.com/plone/plone.schema/issues/6
+
+    2. It does not allow accented characters.
+       Example: jens@österreich.example.com
+       See: https://github.com/plone/plone.schema/issues/9
+
+    3. It does not allow ampersand in the user part.
+       Example: q&a@.example.com
+       See: https://github.com/plone/plone.schema/issues/15
+
+    4. It allows spaces.
+       Example: "test@aha.ok   hello"
+       See: https://github.com/plone/plone.schema/issues/30
+
+    5. It correctly accepts TLDs with more than 4 characters,
+       but this seems by accident, so it could get lost in an update.
+       Example: me@you.online
+       See: https://github.com/plone/plone.schema/issues/30
+    """
+    # We only accept a string.
+    if not isinstance(value, str):
+        return False
+
+    # It is up to the caller to strip spaces, newlines, etc.
+    if value != value.strip():
+        return False
+    if len(value.split()) != 1:
+        return False
+
+    # only one @ sign
+    if value.count("@") != 1:
+        return False
+
+    # At least one dot in the domain.  And when split on dot,
+    # each part must not be empty.
+    user, domain = value.split("@")
+    if not all(domain.partition(".")):
```

Comment:
```suggestion
    if not all(domain.split(".")):
```

AFAIK a email address with a domain but no TLD part like `root@localhost` would be valid mail adresses, at least it's used in server configurations as standard. Also other hosts in the local network could be addressed without a TLD part.